addon/components/stacked-vertical-bar-chart.js

```diff
@@ -0,0 +1,1015 @@
+   * and color.
+   * @type {Array.<Object>}
+   */
+  finishedData: Ember.computed('sortedData', 'otherSliceLabel', function() {
+    var posTop, negBottom, stackedValues, otherSliceLabel;
+    otherSliceLabel = this.get('otherSliceLabel');
+    return _.map(this.get('sortedData'), (values, barLabel) => {
+      posTop = 0;
+      negBottom = 0;
+      stackedValues = _.map(values, function(d) {
+        var yMin, yMax;
+        if (d.value < 0) {
+          yMax = negBottom;
+          negBottom += d.value;
+          yMin = negBottom;
+        } else {
+          yMin = posTop;
+          posTop += d.value;
+          yMax = posTop;
+        }
+        return {
+          yMin: yMin,
+          yMax: yMax,
+          value: d.value,
+          barLabel: d.barLabel,
+          sliceLabel: d.sliceLabel,
+          color: d.color
+        };
+      });
+
+      return {
+        barLabel: barLabel,
+        values: values,
+        stackedValues: stackedValues,
+        max: posTop,
+        min: negBottom
+      };
+    });
+  }),
+
+  // ---------------------------------------------------------------------------
+  // Slice and Bar Sorting
+  // ---------------------------------------------------------------------------
+
+  /**
+   * Key used to determine slice sorting order. Can be 'value', 'original', or
+   * 'other'.
+   * @see valueSliceSortingFn
+   * @see originalSliceSortingFn
+   * @see customSliceSortingFn
+   * @type {string}
+   */
+  sliceSortKey: 'value',
+
+  /**
+   * Slice order for when sliceSortKey is set to `value`
+   * Starting with the largest net-value bar, sort slices in each bar by abs.
+   * value, and add these to the slice order (from largest to smallest)
+   * assuming they are not already in the order. Then repeat this process for
+   * all bars to make sure all slices are listed in the order.
+   * @see sliceSortKey
+   * @see valueSliceSortingFn
+   * @type {Array.<string>}
+   */
+  sliceOrderByValue: Ember.computed('netBarValues.[]', 'dataGroupedByBar', 'otherSliceLabel', function() {
+    var sortedBars, sliceOrder, slicesInBar, allSlicesByBar;
+    allSlicesByBar = this.get('dataGroupedByBar');
+    sortedBars = _.sortBy(this.get('netBarValues'), 'value').reverse();
+    sliceOrder = [];
+    sortedBars.forEach(bar => {
+      slicesInBar = _.sortBy(allSlicesByBar[bar.barLabel], slice => {
+        return -Math.abs(slice.value);
+      });
+      slicesInBar.forEach(slice => {
+        if (sliceOrder.indexOf(slice.sliceLabel) === -1) {
+          sliceOrder.push(slice.sliceLabel);
+        }
+      });
+    });
+    sliceOrder.push(this.get('otherSliceLabel'));
+    return sliceOrder;
+  }),
+
+  /**
+   * Comparison function for slices for when sliceSortKey is 'value'
+   * @see sliceSortKey
+   * @see sliceOrderByValue
+   * @type {function}
+   */
+  valueSliceSortingFn: Ember.computed('sliceOrderByValue.[]', function() {
+    var sliceOrder = this.get('sliceOrderByValue');
```

Comment:
Why do we need extra CP `sliceOrderByValue` instead of directly compare slices' values here?